src/Type/Constant/ConstantFloatType.php

```diff
@@ -45,18 +45,24 @@ public function equals(Type $type): bool
 		return $type instanceof self && abs($this->value - $type->value) < PHP_FLOAT_EPSILON;
 	}
 
+	private function castFloatToString(float $value): string
+	{
+		$precisionBackup = ini_get('precision');
+		ini_set('precision', '-1');
+		try {
+			$valueStr = (string) $value;
+
+			return is_finite($value) && strpos($valueStr, '.') === false ? $valueStr . '.0' : $valueStr;
+		} finally {
+			ini_set('precision', $precisionBackup);
+		}
+	}
+
 	public function describe(VerbosityLevel $level): string
 	{
 		return $level->handle(
 			static fn (): string => 'float',
-			function (): string {
-				$formatted = (string) $this->value;
-				if (is_finite($this->value) && strpos($formatted, '.') === false) {
-					$formatted .= '.0';
-				}
-
-				return $formatted;
-			},
+			fn (): string => $this->castFloatToString($this->value),
```

Comment:
IIRC this is a perf critical path. 

I think we need a benchmark to show whether this edge-case fixing leads to a perf regression in the common case

Reply:
Scratch that. I mixed it up with https://github.com/phpstan/phpstan-src/pull/2080

Reply:
#2080 - please see https://github.com/phpstan/phpstan/issues/9228, the comparison should be even simplifid to `===` only, as any other float value is different